chain/deals/client_utils.go

```diff
@@ -98,3 +109,81 @@ func (c *Client) disconnect(deal ClientDeal) error {
 	delete(c.conns, deal.ProposalCid)
 	return err
 }
+
+var _ datatransfer.RequestValidator = &ClientRequestValidator{}
+
+// ClientRequestValidator validates data transfer requests for the client
+// in a storage market
+type ClientRequestValidator struct {
+	deals *statestore.StateStore
+}
+
+// RegisterClientValidator is an initialization hook that registers the client
+// request validator with the data transfer module as the validator for
+// StorageDataTransferVoucher types
+func RegisterClientValidator(lc fx.Lifecycle, crv *ClientRequestValidator, dtm datatransfer.ClientDataTransfer) {
+	lc.Append(fx.Hook{
+		OnStart: func(ctx context.Context) error {
+			return dtm.RegisterVoucherType(reflect.TypeOf(StorageDataTransferVoucher{}), crv)
```

Comment:
Can this registration just be done in the constructor below? I'd like to avoid importing fx into non-builder packages

Reply:
Sorry moved from #500:

It could be in the constructor I guess. The validator doesn't otherwise contain any reference to the DT module. This also provides a way to make sure the validator gets instantiated. (cause no other modules depend on it ATM).

I'm under the impression (perhaps incorrect) that FX doesn't instantiate modules if they aren't depended on something that's in a lifecycle hook or instantiated directly by top level code.

@magik6k has been working with FX a lot longer than me so perhaps he can speak to this.

Reply:
what about just moving this code?